packages/mui-joy/src/Box/Box.test.js

```diff
@@ -65,4 +65,189 @@ describe('Joy <Box />', () => {
+          <Box sx={{ backgroundColor: 'primary.500' }} />
+        </CssVarsProvider>,
+      );
+
+      expect(container.firstChild).toHaveComputedStyle({
+        backgroundColor: 'rgb(0, 0, 255)',
+      });
+    });
+
+    it('borderRadius', function test() {
+      const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+      if (isJSDOM) {
+        this.skip();
+      }
+
+      const { container } = render(
+        <CssVarsProvider theme={theme}>
+          <Box sx={{ borderRadius: 'md' }} />
+        </CssVarsProvider>,
+      );
+
+      expect(container.firstChild).toHaveComputedStyle({
+        borderTopLeftRadius: '77px',
+        borderTopRightRadius: '77px',
+        borderBottomLeftRadius: '77px',
+        borderBottomRightRadius: '77px',
+      });
+    });
+
+    it('boxShadow', function test() {
+      const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+      if (isJSDOM) {
+        this.skip();
+      }
+
+      const { container } = render(
+        <CssVarsProvider theme={theme}>
+          <Box sx={{ boxShadow: 'md' }} />
+        </CssVarsProvider>,
+      );
+
+      expect(container.firstChild).toHaveComputedStyle({
+        boxShadow: 'rgb(0, 0, 0) 0px 0px 10px 0px',
+      });
+    });
+
+    it('fontSize', function test() {
+      const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+      if (isJSDOM) {
+        this.skip();
+      }
+
+      const { container } = render(
+        <CssVarsProvider theme={theme}>
+          <Box sx={{ fontSize: 'md' }} />
+        </CssVarsProvider>,
+      );
+
+      expect(container.firstChild).toHaveComputedStyle({
+        fontSize: '16px',
+      });
+    });
+
+    it('fontWeight', function test() {
+      const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+      if (isJSDOM) {
+        this.skip();
+      }
+
+      const { container } = render(
+        <CssVarsProvider theme={theme}>
+          <Box sx={{ fontWeight: 'md' }} />
+        </CssVarsProvider>,
+      );
+
+      expect(container.firstChild).toHaveComputedStyle({
+        fontWeight: '500',
+      });
+    });
+
+    it('letterSpacing', function test() {
+      const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+      if (isJSDOM) {
+        this.skip();
+      }
+
+      const { container } = render(
+        <CssVarsProvider theme={theme}>
+          <Box sx={{ letterSpacing: 'md' }} />
+        </CssVarsProvider>,
+      );
+
+      expect(container.firstChild).toHaveComputedStyle({
+        letterSpacing: '1.328px',
+      });
+    });
+
+    it('lineHeight', function test() {
+      const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+      if (isJSDOM) {
+        this.skip();
+      }
+
+      const { container } = render(
+        <CssVarsProvider theme={theme}>
+          <Box sx={{ lineHeight: 'md' }} />
+        </CssVarsProvider>,
+      );
+
+      expect(container.firstChild).toHaveComputedStyle({
+        lineHeight: '24px',
+      });
+    });
```

Comment:
Since we have added quite a number of these tests, I wonder if there is an opportunity with lower-level tests. I'm not sure, only that not having JSDOM can make iterating on tests a bit slower.

Reply:
Honestly, I would be against adding tests for internals (as that is what `styleFunctionSx` is, and rather relay on the `sx` prop to work as expected, as that is the goal). As we've seen now, changing the internals required writing different tests, which is not great. So, having this said, I would ultimately want to avoid it next time we need to do some internal changes (maybe update the system to use utility classes :D)

Reply:
Agree 👍